building_map_tools/building_crowdsim/config/plugin_file.py

```diff
@@ -0,0 +1,175 @@
+            self.attributes['filename'] = 'libcrowd_simulator_ign.so'
+        else:
+            raise ValueError(
+                "Unknown 'model_env' provided to initialize Plugin: [" +
+                model_env + "]")
+
+        self.model_env = model_env
+        self.attributes['name'] = 'crowd_simulation'
+
+        self.behavior_file_element = LeafElement('behavior_file')
+        self.behavior_file_element.text = 'behavior_file.xml'
+
+        self.scene_file_element = LeafElement('scene_file')
+        self.scene_file_element.text = 'scene_file.xml'
+
+        self.update_time_step_element = LeafElement('update_time_step')
+        self.update_time_step_element.text = '0.1'
+
+        self.sub_elements.append(self.behavior_file_element)
+        self.sub_elements.append(self.scene_file_element)
+        self.sub_elements.append(self.update_time_step_element)
+
+    def load_from_yaml(self, yaml_node):
+        if 'enable' not in yaml_node:
+            raise ValueError("Missing 'enable' in yaml_node")
+        if 'update_time_step' not in yaml_node:
+            raise ValueError("Missing 'update_time_step' in yaml_node")
+        self.update_time_step_element.text = str(
+            float(yaml_node['update_time_step']))
+
+        if 'model_types' not in yaml_node:
+            raise ValueError("Invalid 'model_types' in yaml_node")
+        for model_type in yaml_node['model_types']:
+            cur_model_type = ModelType()
+            cur_model_type.load_from_yaml(model_type)
+            self.add_model_type(cur_model_type)
+
+        # get all the external agent names from 'agent_groups'
+        if 'agent_groups' not in yaml_node:
+            return
+        agent_groups = yaml_node['agent_groups']
+        for group in agent_groups:
+            if len(group['agents_name']) == 0:
+                continue
+            self.add_external_list(group['agents_name'])
+
+    def add_model_type(self, model_type):
+        assert(isinstance(model_type, ModelType))
+        model_type.model_env = self.model_env
+        self.sub_elements.append(model_type)
+
+    def add_external_agent(self, name):
+        external_agent_element = LeafElement('external_agent')
+        external_agent_element.text = name
+        self.sub_elements.append(external_agent_element)
+
+    def add_external_list(self, name_list):
+        for name in name_list:
+            self.add_external_agent(name)
+
+
+class ModelType (Element):
+    def __init__(self):
+        Element.__init__(self, 'model_type')
+        self.type_name = LeafElement('typename')
+        self.animation_speed = LeafElement('animation_speed')
+        self.animation = LeafElement('animation')
+        # for gazebo model
+        self.initial_pose_gazebo = LeafElement('initial_pose')
+        self.gazebo_filename = LeafElement('filename')
+        # for ign model
+        self.initial_pose_ign = LeafElement('initial_pose')
+        self.ign_filename = LeafElement('filename')
```

Comment:
This is a bit tricky. Because currently, to spawn an actor, gazebo requires to provide .dae file like
`model://MaleVisitorPhone/meshes/MaleVisitorPhoneWalk.dae`. If the idle animation is provided, another idle.dae should be provided. Ignition only needs a http dir like `https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Male%20visitor` or `model://MaleVisitorPhone` . The initial pose can be suppressed though if all the models exsits locally. What's your opinion on this?